keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/multiple/ProcessModelVertexTest.java

```diff
@@ -0,0 +1,239 @@
+package io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.multiple;
+
+import io.improbable.keanu.algorithms.NetworkSamples;
+import io.improbable.keanu.algorithms.mcmc.MetropolisHastings;
+import io.improbable.keanu.algorithms.variational.optimizer.nongradient.NonGradientOptimizer;
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.tensor.Tensor;
+import io.improbable.keanu.tensor.bool.BooleanTensor;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.intgr.IntegerTensor;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.VertexLabel;
+import io.improbable.keanu.vertices.bool.BoolVertex;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.ConstantDoubleVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+import io.improbable.keanu.vertices.intgr.IntegerVertex;
+import io.improbable.keanu.vertices.model.LambdaModelVertex;
+import io.improbable.keanu.vertices.model.ModelVertex;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+public class ProcessModelVertexTest {
+
+    /*
+    The model we are mimicking here is a python script, model.py.
+
+    It takes one input, Temperature, and produces two outputs of type double, Chance of Rain & Humidity. These outputs
+    are written to file.
+
+    It also produces one integer output and one boolean output. Suggested Factor of Suncream and 'is it sunny'. These
+    are also written to file.
+     */
+
+    private KeanuRandom random;
+    private DoubleVertex inputToModel;
+    private SimpleWeatherModel weatherModel;
+
+    @Before
+    public void mockFilesToReadModelOutputFrom() throws IOException {
+        random = new KeanuRandom(1);
+        weatherModel = new SimpleWeatherModel(inputToModel);
+    }
+
+    @Test
+    public void canRunAModelInAModel() {
+        inputToModel = new ConstantDoubleVertex(25);
+        weatherModel.setInputToModel(inputToModel);
```

Comment:
ditto my earlier comment - set `inputToModel` in a `@Before` method and you won't need these 2 lines

Reply:
Nice changed